crypto/storage/client.go

```diff
@@ -0,0 +1,109 @@
+package storage
+
+import (
+	"context"
+	"crypto"
+	"fmt"
+	"github.com/nuts-foundation/nuts-node/crypto/storage/httpclient"
+	"github.com/nuts-foundation/nuts-node/crypto/util"
+	"net/http"
+)
+
+// APIClient implements the Storage interface. It uses a simple HTTP protocol to connect to an external storage server.
+// This server can either be a secret store itself, or proxy the request to a key store such as Hashicorp Vault or Azure Key Vault.
+// It allows us to keep the codebase clean and allow other parties to write their own adaptor.
+type APIClient struct {
+	httpClient *httpclient.ClientWithResponses
+}
+
+type APIClientConfig struct {
+	URL string `koanf:"url"`
+}
+
+func NewAPIClient(url string) (Storage, error) {
+	client, _ := httpclient.NewClientWithResponses(url)
+	response, err := client.HealthCheckWithResponse(context.Background())
+	if err != nil {
+		return nil, fmt.Errorf("unable to connect to storage server: %w", err)
+	}
+	if response.StatusCode() != http.StatusOK {
+		return nil, fmt.Errorf("unable to connect to storage server: unexpected status code: %d", response.StatusCode())
+	}
+
+	return &APIClient{httpClient: client}, nil
+}
+
+type backendError struct {
+	error httpclient.ErrorResponse
+}
+
+func (r backendError) Error() string {
+	return fmt.Sprintf("remote error: the backend %s returned an error with status %d, title=%s details=%s", r.error.Backend, r.error.Status, r.error.Title, r.error.Detail)
```

Comment:
We should take care this is not returned to clients, since it potentially leaks sensitive information. Maybe just log the backend/status/details and not return it in an error the calling HTTP client.

Might be a good point to document in the implementation guide (what to return, and what specifically not)